src/uct/ib/rc/base/rc_ep.h

```diff
@@ -353,7 +351,8 @@ uct_rc_txqp_completion_inl_resp(uct_rc_txqp_t *txqp, const void *resp, uint16_t
 static UCS_F_ALWAYS_INLINE uint8_t
 uct_rc_iface_tx_moderation(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp, uint8_t flag)
 {
-    return (txqp->unsignaled >= iface->config.tx_moderation) ? flag : 0;
+    return ((txqp->unsignaled >= iface->config.tx_moderation) ||
```

Comment:
how about to set `txqp->unsignaled` to `inf` in `uct_rc_txqp_posted`?
`if (ucs_unlikely(iface->tx.cq_available -= res_count) <= 0)) {
    txqp->unsignaled_real = txqp->unsignaled;
    txqp->unsignaled = UINT16_MAX;
}`
this check in the same scope on cached value should be pretty fast,  also we need to check this value on completion to replace `inf` value of with saved real value (in the same cache line) with `unsignaled`  in "bad case"

On the next iteration post will be definitely "signaled" and the value will be initialized with zero as usual.